src/components/Router.tsx

```diff
@@ -1,34 +1,54 @@
 import Dashboard from 'components/Dashboard'
 import Landing from 'components/Landing'
-import { HashRouter, Route, Switch } from 'react-router-dom'
+import { ChildElems } from 'models/child-elems'
+import { useHistory } from 'react-router'
+import { BrowserRouter, HashRouter, Route, Switch } from 'react-router-dom'
+import { isCloudflare } from 'utils/cloudflare'
 
 import CatchallRedirect from './CatchallRedirect'
 import Create from './Create'
 import Projects from './Projects'
 
-export default function Router() {
+interface RouterProps {
+  children: ChildElems
+}
+
+export const Router = ({ children }: RouterProps) => {
+  const history = useHistory()
+
+  if (isCloudflare()) {
+    // Redirects any hash routes to regular routes
+    if (window && window.location.hash.startsWith('#/')) {
+      history.push(window.location.hash.replace('#', ''))
+    }
+    // Use regular routes
+    return <BrowserRouter>{children}</BrowserRouter>
+  }
+
+  return <HashRouter>{children}</HashRouter>
+}
+
+export default function Routes() {
```

Comment:
Do you think we should keep this or just move completely into BrowserRouter? I favor the BrowserRouter and I can't see a use-case where hash routing gives us any perceptible benefits.

Reply:
the hash routing is necessary for apps served on IPFS. without the hash, IPFS will try to load a file at the subroute (i.e. juicebox.money/subroute) instead of at the root, which is what we need to load the SPA

Reply:
Ah, I see.

Reply:
@torvusbug if we ever decide to move to next/vercel, we should revisit this.